stratosphere/spl/source/spl_types.hpp

```diff
@@ -21,141 +21,136 @@
-    SmcResult_NotImplemented = 1,
-    SmcResult_InvalidArgument = 2,
-    SmcResult_InProgress = 3,
-    SmcResult_NoAsyncOperation = 4,
-    SmcResult_InvalidAsyncOperation = 5,
-    SmcResult_Blacklisted = 6,
-
-    SmcResult_Max = 99,
-};
-
-enum SmcCipherMode : u32 {
-    SmcCipherMode_CbcEncrypt = 0,
-    SmcCipherMode_CbcDecrypt = 1,
-    SmcCipherMode_Ctr = 2,
-};
-
-enum SmcDecryptOrImportMode : u32 {
-    SmcDecryptOrImportMode_DecryptRsaPrivateKey = 0,
-    SmcDecryptOrImportMode_ImportLotusKey = 1,
-    SmcDecryptOrImportMode_ImportEsKey = 2,
-    SmcDecryptOrImportMode_ImportSslKey = 3,
-    SmcDecryptOrImportMode_ImportDrmKey = 4,
-};
-
-enum SmcSecureExpModMode : u32 {
-    SmcSecureExpModMode_Lotus = 0,
-    SmcSecureExpModMode_Ssl = 1,
-    SmcSecureExpModMode_Drm = 2,
-};
-
-enum EsKeyType : u32 {
-    EsKeyType_TitleKey = 0,
-    EsKeyType_ElicenseKey = 1,
-};
-
-struct AsyncOperationKey {
-    u64 value;
-};
-
-struct BootReasonValue {
-    u8 power_intr;
-    u8 rtc_intr;
-    u8 nv_erc;
-    u8 boot_reason;
-};
-static_assert(sizeof(BootReasonValue) == sizeof(u32), "BootReasonValue definition!");
-
-#pragma pack(push, 1)
-struct AesKey {
-    union {
-        u8 data[AES_128_KEY_SIZE];
-        u64 data64[AES_128_KEY_SIZE / sizeof(u64)];
+namespace sts::spl {
+
+    namespace smc {
+
+        enum class FunctionId : u32 {
+            SetConfig                     = 0xC3000401,
+            GetConfig                     = 0xC3000002,
+            CheckStatus                   = 0xC3000003,
+            GetResult                     = 0xC3000404,
+            ExpMod                        = 0xC3000E05,
+            GenerateRandomBytes           = 0xC3000006,
+            GenerateAesKek                = 0xC3000007,
+            LoadAesKey                    = 0xC3000008,
+            CryptAes                      = 0xC3000009,
+            GenerateSpecificAesKey        = 0xC300000A,
+            ComputeCmac                   = 0xC300040B,
+            ReEncryptRsaPrivateKey        = 0xC300D60C,
+            DecryptOrImportRsaPrivateKey  = 0xC300100D,
+
+            SecureExpMod                  = 0xC300060F,
+            UnwrapTitleKey                = 0xC3000610,
+            LoadTitleKey                  = 0xC3000011,
+            UnwrapCommonTitleKey          = 0xC3000012,
+
+            /* Deprecated functions. */
+            ImportEsKey                   = 0xC300100C,
+            DecryptRsaPrivateKey          = 0xC300100D,
+            ImportSecureExpModKey         = 0xC300100E,
+        };
+
+        enum class Result {
+            Success               = 0,
+            NotImplemented        = 1,
+            InvalidArgument       = 2,
+            InProgress            = 3,
+            NoAsyncOperation      = 4,
+            InvalidAsyncOperation = 5,
+            Blacklisted           = 6,
+
+            Max                   = 99,
+        };
+
+        inline ::Result ConvertResult(Result result) {
```

Comment:
Is `` ::Result`` intentional?

Reply:
Yes, because it returns a Result, not an smc::Result.